### PR TITLE
Make Travis run tests on OSX.

### LINKED ISSUE #368: Mac OS CI
https://github.com/travis-ci/travis-ci/issues/2312
http://docs.travis-ci.com/user/multi-os/

So I think this should be possible from what I've read so far.  Python isn't officially supported on travis with mac os, but there's lots of workarounds that people have used. 

The first few files in this PR show how you can have separate install recipes for linux/mac os
https://github.com/catkin/catkin_tools/pull/196/files

### PR DESCRIPTION
**Pull request by Endle**
Fixes #368 

Changes made in this Pull Request:
 - Make Travis CI run the tests on OSX

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

Signed-off-by: Zhenbo Li <litimetal@gmail.com>

### CODE REVIEW

**Comment by richardjgowers** [comment]
Hey, thanks for taking this on.

The travis check isn't showing up though?

**Comment by Endle** [comment]
Thank you for reviewing my patch.

I'm wondering why this PR hasn't be shown in <https://travis-ci.org/MDAnalysis/mdanalysis/pull_requests>. Despite of it, I've tested my patch on Travis: <https://travis-ci.org/Endle/mdanalysis/builds/116079307>

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,9 @@ branches:
     - master
     - develop
 
-language: python
```
Comment (richardjgowers) [inline .travis.yml]: maybe this line is needed?

File: .travis.yml [hunk 2]
```diff
@@ -13,29 +15,29 @@ env:
     - GIT_CI_USER: TravisCI
     - GIT_CI_EMAIL: TravisCI@mdanalysis.org
     - MDA_DOCDIR: package/doc/html
+  matrix:
```
Comment (richardjgowers) [inline .travis.yml]: Or maybe declaring matrix twice breaks it?

File: .travis.yml (hunk 1 above)
Comment (Endle) [inline .travis.yml]: Python + OSX will break Travis CI.
As we use miniconda, it's fine to skip the language.

Maybe the reason is that I have tested this patch with my repo?

**Comment by kain88-de** [comment]
does travis support multi OS testing by default now? I thought that they require us to sign up for that separately.

**Comment by richardjgowers** [comment]
Iirc osx is supported, but not Python on osx. But conda works around that

File: .travis.yml (hunk 2 above)
Comment (kain88-de) [inline .travis.yml]: Yeah that is a very good chance. I'm not aware of duplicate keys in the YAML standard.

**Comment by richardjgowers** [comment]
@Endle this commit seems to get travis working

https://github.com/MDAnalysis/mdanalysis/pull/773/commits/299b243fa19db46ea88ad5154dd4b141ffb649c7

But (for now) it looks like it only starts 2 jobs?

**Comment by kain88-de** [comment]
> But (for now) it looks like it only starts 2 jobs?

now it started all 8 defined jobs for this PR

**Comment by richardjgowers** [comment]
Cool, thanks @Endle !

**Pull request merged**